### ISSUE TITLE
UCSB domain name

### ISSUE THREAD

**Issue by 392781** [UCSBCarpentry/ucsbcarpentry.github.io#22]
I was wondering if UCSB Carpentry has considered getting IT to give it a UCSB domain name?

Something like `carpentry.ucsb.edu`?  Or `carpentry.library.ucsb.edu`?  The benefits of this is having an official UCSB domain instead of relying for one under `github.io`. 

The process is very straighforward:
* Contact IT to request a subdomain like the 2 suggested above.
* Request them to create a `CNAME` entry in their DNS that points `ucsbcarpentry.github.io` to the new domain.
* Create a file called `CNAME` in the root of this repository containing just the domain name. Example `CNAME` contents:

```
carpentry.ucsb.edu
```

* Make sure that "Enforce HTTPS" is enabled (often unnecessary since GitHub Pages seems to do this automatically most of the time): https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https

More information about the gist of this can be found here: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site

**Comment by ilessing** [comment]
If we decide we want `carpentry.library.ucsb.edu` the approval process is within the Library's IT organization and is typically pretty easy.

If we decide we want `carpentry.ucsb.edu` the approval process is a bit more involved.  An informal request for comment gets circulated and then any unit on campus can object.

**Comment by 392781** [comment]
I want to bring this back @josenino95 @srerickson. I think it would be highly beneficial to get a formal URL for the website. If there are issues with getting a `ucsb.edu` one, domain names are pretty cheap (so much so that I wouldn't mind paying for one myself in this case indefinitely and have done so for other orgs).

**Comment by josenino95** [comment]
Thanks for bringing this back @392781 ! Today, we talked about it in a DREAM Lab meeting, and we are moving forward to with carpentry.library.ucsb.edu. Seth and I will go through the process with IT. I'll keep you updated.

**Comment by ilessing** [comment]
bringing this to the attention of @michael-cr41g

**Comment by 392781** [comment]
Awww yeah!!! 🥳

**Comment by josenino95** [comment]
Hi @chrissyrissmeyer. Yes, Amber was consulted on this matter. I just submitted the ticket. Thank you!

**Comment by josenino95** [comment]
@ilessing @michael-cr41g I followed @chrissyrissmeyer recommendation of submitting the DNS Change Request ticket, but I still got a response that I should reach out to Library IT. Here is the complete answer I received:

> Subdomains (any *.library.ucsb.edu) do not need campus' approval, provided top level domain already approved or exists. Please note that DNS zone "library.ucsb.edu" is delegated to Route53, thus for any *.library.ucsb.edu records contact Library IT department.
> 
> Igor Shabaltas
> Campus Network Programmer
> Information Technology Services, Network & Communications
> University of California, Santa Barbara

Do you have suggestions on who I should contact to move forward with this?

**Comment by ilessing** [comment]
apologies we should not have sent you to campus IT for the name:

- `carpentry.library.ucsb.edu`

can you confirm that you want the Github pages website currently at https://ucsbcarpentry.github.io/  to be available at: https://carpentry.library.ucsb.edu/  ??

@michael-cr41g I think we'll need to follow instructions at [Managing a custom domain for your GitHub Pages site - GitHub Docs](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain)

**Comment by josenino95** [comment]
Yes, I can confirm that's the change we want. Thank you @ilessing and @michael-cr41g!

**Comment by ilessing** [comment]
This is done!  yay!!!

![Image](https://github.com/user-attachments/assets/fe328e32-3028-45c0-b90e-aaeb7af1b3ce)

![Image](https://github.com/user-attachments/assets/3ca02863-852a-472a-a278-7e2a41a6cfab)

@josenino95  please close this issue when you've tested and approve.

**Comment by 392781** [comment]
@ilessing @josenino95 tests fine and is the final stretch but not quite done as there is no SSL certificate so the site provides only an http connection. Https displays an insecure connection error. Https would be nice to have to conform to modern web standards.

**Comment by 392781** [comment]
Oops, just noticed this was 4 mins ago. Will probably be updated soon.

**Comment by 392781** [comment]
HTTPS functions! New URL works great. Closing this issue as complete :^)

**Comment by srerickson** [comment]
@ilessing I think there is a DNS issue that only affects computers in the library and prevents accessing the site. From within the library, computers can't resolve the hostname `carpentry.library.ucsb.edu`. Outside the library network it works fine. That's the shape of the issue as it appears to me now. Any ideas?

**Comment by josenino95** [comment]
Great, I'll do that. Thank you for all your help with this!

**Comment by ilessing** [comment]
```
From: Jose Nino Muriel 
Date: Thu, Mar 20, 2025 at 1:26 PM
Subject: Re: carpentry.library.ucsb.edu
To: Ian Lessing
Hi Ian,
I've tested it and now it works perfectly. Thank you both!
Jose
```

---
